src/app/time.service.spec.ts

```diff
@@ -0,0 +1,33 @@
+import {TestBed} from '@angular/core/testing';
+
+import {TimeService} from './time.service';
+
+import * as moment from 'moment';
+
+describe('TimeService', () => {
+  let service: TimeService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(TimeService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
```

Comment:
for (const [dateString, expectedTime] of ['noon date strign', 'noon'], [...]) {
  it(`should get correct time-of-day at ${expectedTime}`, () => {
    const expectedTime = moment(dateString);
    spyOn(service, 'moment') // mock moment with expected time
    expect(service.timeOfDay()).toEqual(expectedTime);
  });
}